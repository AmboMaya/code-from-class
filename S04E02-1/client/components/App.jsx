import React from 'react'

class App extends React.Component {
  state = {
    visible: true,
    colour: 'green'
  }

  toggleVisibility = () => {
    this.setState({
      visible: false,
      colour: 'blue'
    })
  }

  render () {
    return (
      <div>
        <button onClick={this.toggleVisibility}>HIDE ME</button>
        {this.state.visible && <div>MY APP</div>}
        <Child />
      </div>
    )
  }
}

class Child extends React.Component {
  handleMouseMove = () => {
    console.log('THE MOUSE IS MOVING OVER THIS COMPONENT YIKES')
  }

  render () {
    return (
      <div onMouseMove={this.handleMouseMove}>
        Hi!
      </div>
    )
  }
}

export default App

