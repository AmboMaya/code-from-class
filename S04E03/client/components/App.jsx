import React from 'react'
import { HashRouter as Router, Route } from 'react-router-dom'

import Home from './Home'
import Users from './Users'

const App = () => {
  return (
    <Router>
      <div>
        <Route exact path='/' component={Home} />
        <Route path='/users/:id' component={Users} />
      </div>
    </Router>
  )
}

export default App

