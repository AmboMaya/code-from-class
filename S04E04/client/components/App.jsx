import React from 'react'

const MIN_USERNAME_LENGTH = 5

class App extends React.Component {
  // constructor (props) {
  //   super(props)
  //   this.state = {}
  // }

  state = {
    message: '',
    password: '',
    username: ''
  }

  changeHandler = e => {
    let message = ''
    if (e.target.name === 'username' && e.target.value.length < MIN_USERNAME_LENGTH) {
      message = 'That username is too short!'
    }
    this.setState({
      [ e.target.name ]: e.target.value,
      message
    })
  }

  submitHandler = e => {
    e.preventDefault()

    // Final validation before submitting to server
    console.log('SUBMITTED:', this.state.username, this.state.password)
  }

  render () {
    return (
      <div>
        <form onSubmit={this.submitHandler}>
          <input
            type='text'
            value={this.state.username}
            onChange={this.changeHandler}
            placeholder='Username'
            name='username' />
          <input
            type='password'
            value={this.state.password}
            onChange={this.changeHandler}
            placeholder='Password'
            name='password' />
          <input
            type='submit'
            value='Log in!' />
        </form>
        <p>{this.state.message}</p>
      </div>
    )
  }
}

export default App

