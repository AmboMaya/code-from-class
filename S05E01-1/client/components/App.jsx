import React from 'react'
import request from 'superagent'

class App extends React.Component {
  state = {
    message: ''
  }

  componentDidMount () {
    request
      .get('/message')
      .then(res => {
        // We can only use the API data here
        this.setState({ message: res.body.message })
      })
      .catch(err => {
        // Error messages will be available here
        console.error(err)
      })
  }

  render () {
    return (
      <React.Fragment>
        <h1>The message was:</h1>
        <p>{this.state.message}</p>
      </React.Fragment>
    )
  }
}

export default App

