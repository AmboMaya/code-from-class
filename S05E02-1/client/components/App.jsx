import React from 'react'

import AddWidget from './AddWidget'
import EditWidget from './EditWidget'
import ErrorMessage from './ErrorMessage'
import WidgetDetails from './WidgetDetails'
import WidgetList from './WidgetList'

import * as api from '../api'

export default class App extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      activeWidget: null,
      addWidgetVisible: false,
      detailsVisible: false,
      editWidget: null,
      error: null,
      widgets: []
    }

    this.deleteWidget = this.deleteWidget.bind(this)
    this.editWidget = this.editWidget.bind(this)
    this.hideDetails = this.hideDetails.bind(this)
    this.refreshList = this.refreshList.bind(this)
    this.showAddWidget = this.showAddWidget.bind(this)
    this.showDetails = this.showDetails.bind(this)
  }

  componentDidMount () {
    this.refreshList()
  }

  refreshList () {
    api.getWidgets()
      .then(widgets => {
        this.setState({
          error: null,
          addWidgetVisible: false,
          detailsVisible: false,
          editWidget: null,
          widgets
        })
      })
      .catch(error => this.setState({ error }))
  }

  showAddWidget () {
    this.setState({
      addWidgetVisible: true,
      detailsVisible: false
    })
  }

  showDetails (widget) {
    this.setState({
      activeWidget: widget,
      detailsVisible: true
    })
  }

  hideDetails () {
    this.setState({
      detailsVisible: false
    })
  }

  deleteWidget (id) {
    api.deleteWidget(id)
      .then(this.refreshList)
      .catch(error => this.setState({ error }))
  }

  editWidget (id) {
    this.setState({
      editWidget: id
    })
  }

  render () {
    return (
      <div>
        <ErrorMessage error={this.state.error} />

        <h1>Widgets FTW!</h1>

        <WidgetList
          editWidget={this.editWidget}
          showDetails={this.showDetails}
          widgets={this.state.widgets} />

        <p>
          <a id='show-widget-link' href='#'
            onClick={this.showAddWidget}>Add widget</a>
        </p>

        {this.state.addWidgetVisible && <AddWidget
          refresh={this.refreshList} />}

        {this.state.editWidget && <EditWidget
          refresh={this.refreshList}
          widgetId={this.state.editWidget} />}

        {this.state.detailsVisible && <WidgetDetails
          deleteWidget={this.deleteWidget}
          hideDetails={this.hideDetails}
          isVisible={this.state.detailsVisible}
          widget={this.state.activeWidget} />}
      </div>
    )
  }
}
