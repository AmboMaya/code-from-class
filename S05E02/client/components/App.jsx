import React from 'react'
import request from 'superagent'

class App extends React.Component {
  state = {
    message: ''
  }

  componentDidMount () {
    request
      .post('http://localhost:3000')
      .send({
        greeting: 'Hi, please accept my POST request!'
      })
      .then(res => {
        this.setState({ message: res.body.message })
      })
      .catch(err => {
        this.setState({ message: err.message })
      })
  }

  render () {
    return (
      <React.Fragment>
        <h1>The server said:</h1>
        <h2>{this.state.message}</h2>
      </React.Fragment>
    )
  }
}

export default App

