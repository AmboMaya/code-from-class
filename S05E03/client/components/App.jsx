import React from 'react'
import request from 'superagent'

class App extends React.Component {
  state = {
    spaceX: {
      apiURL: 'https://api.spacexdata.com/v3/launches',
      missions: [],
      latestMission: ''
    },
    weather: {
      apiURL: 'http://api.apixu.com/v1/current.json',
      key: '3ab2118b64934865a1b55006181311',
      city: 'Auckland',
      tempInC: null
    }
  }
  
  componentDidMount () {
    this.getSpaceXData()
    this.getWeatherData()
  }

  getSpaceXData () {
    const {apiURL} = this.state.spaceX

    request.get(apiURL)
      .then(res => {
        this.setState({
          spaceX: {
            missions: res.body,
            latestMission: this.getLatestMission(res.body)
          }
        })
      })
  }

  getWeatherData () {
    const {apiURL, key, city} = this.state.weather

    request.get(apiURL + '?key=' + key + '&q=' + city)
      .then(res => {
        this.setState({
          weather: {
            ...this.state.weather,
            tempInC: res.body.current.temp_c + 'Â°C'
          }
        })
      })
  }

  getLatestMission (missions) {
    return missions.find(mission => mission.upcoming)
  }

  render () {
    return (
      <React.Fragment>
        <h1>Next SpaceX mission:</h1>
        <p>{this.state.spaceX.latestMission.mission_name}</p>
        <br />
        <h1>Weather in {this.state.weather.city}:</h1>
        <p>{this.state.weather.tempInC}</p>
      </React.Fragment>
    )
  }
}

export default App
