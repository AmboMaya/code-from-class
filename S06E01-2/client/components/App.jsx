import React from 'react'

import Wombats from './Wombats'

function App (props) {
  const wombats = props.wombats
  return (
    <div>
      <Wombats dispatch={props.dispatch} wombats={wombats} />
    </div>
  )
}

export default App
