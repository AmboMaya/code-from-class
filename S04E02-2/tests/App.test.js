import React from 'react'
import { mount, shallow } from 'enzyme'
import render from 'react-test-renderer'

import App from '../client/components/App'
import List from '../client/components/List'

test('<App />', () => {
  const expected = true
  const wrapper = shallow(<App />)
  const actual = wrapper.containsMatchingElement(List)
  expect(actual).toBe(expected)
})

test('<List /> calls moreWombatInfo when item clicked', () => {
  // Arrange
  const mockCallback = jest.fn()
  const wombats = ['herschel']
  const expected = 1
  const component = <List things={wombats} moreWombatInfo={mockCallback} />

  // Act
  const wrapper = mount(component)
  const li = wrapper.find('li').first()
  li.simulate('click')
  const actual = mockCallback.mock.calls.length

  // Assert
  expect(actual).toBe(expected)
})

test('<App /> matches the last snapshot', () => {
  const actual = render.create(<App />).toJSON()
  expect(actual).toMatchSnapshot()
})
