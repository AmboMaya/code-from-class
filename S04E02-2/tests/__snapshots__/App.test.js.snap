// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the last snapshot 1`] = `
<div>
  <h1>
    React development has begun...
  </h1>
  <div>
    <li
      onClick={[Function]}
    >
      herschel
    </li>
    <li
      onClick={[Function]}
    >
      gertrude
    </li>
    <li
      onClick={[Function]}
    >
      jemima
    </li>
    <div
      className="notes"
    >
      This list of wombats may not be exhaustive.
    </div>
  </div>
</div>
`;
